src/commands/createFunction/dotnetSteps/DotnetFunctionCreateStep.ts

```diff
@@ -53,6 +56,32 @@ export class DotnetFunctionCreateStep extends FunctionCreateStepBase<IDotnetFunc
         }
         await executeDotnetTemplateCommand(context, version, projectTemplateKey, context.projectPath, 'create', '--identity', template.id, ...args);
 
+        await this._installDurableDependenciesIfNeeded(context);
+
         return path.join(context.projectPath, functionName + getFileExtension(context));
     }
+
+    private async _installDurableDependenciesIfNeeded(context: IDotnetFunctionWizardContext): Promise<void> {
+        if (!context.newDurableStorageType) {
+            return;
+        }
+
+        try {
+            switch (context.newDurableStorageType) {
+                case DurableBackend.Netherite:
+                    await cpUtils.executeCommand(ext.outputChannel, context.projectPath, 'dotnet', 'add', 'package', durableUtils.dotnetDfNetheritePackage);
+                    break;
+                case DurableBackend.SQL:
+                    await cpUtils.executeCommand(ext.outputChannel, context.projectPath, 'dotnet', 'add', 'package', durableUtils.dotnetDfSqlPackage);
+                    break;
+                case DurableBackend.Storage:
+                default:
+            }
+
+            // Seems that the package arrives out-dated and needs to be updated
+            await cpUtils.executeCommand(ext.outputChannel, context.projectPath, 'dotnet', 'add', 'package', durableUtils.dotnetDfBasePackage);
+        } catch {
+            ext.outputChannel.appendLog(localize('funcInstallFailed', 'WARNING: Failed to install and update Durable Functions NuGet packages to the .csproj project file. You may need to configure them manually or start from a clean project.'))
```

Comment:
Hmm good question, I'm open to either, but my initial thought is that we probably don't want to throw an error and halt the whole process.  I was considering the package install to be more of a 'nice-to-have' and not a 'need-to-have' enhancement since often times there are instructions in the template files guiding the user on how to install durable function dependencies.  

Your comment did make me re-consider how useful the current log message is to the user though, so I ended up re-coding it to list out all the packages that still need to be installed rather than just the all-in-one generic message that I'm using here, let me know if you feel like additional changes should be made or if you prefer the error message instead!